subcommand/acl-controller/command_test.go

```diff
@@ -232,69 +259,25 @@ func checkConsulResources(t *testing.T, consulClient *api.Client, expPolicyRules
 		policyNames = append(policyNames, policy.Name)
 	}
 
-	require.Contains(t, policyNames, "consul-ecs-client-policy")
 	if partitionsEnabled {
 		// We test with a non-default partition which lacks the global-management policy.
 		// The anonymous token policy is only created in the default partition, since that
-		// is where the anonymous token lives, so we expect only the client policy.
-		require.Len(t, policies, 1)
+		// is where the anonymous token lives, so we expect no policies to be present
+		require.Len(t, policies, 0)
```

Comment:
minor/unrelated maintenance thing that is not your fault: We are adding a new built-in policy to Consul in the near future, and it is likely to go into 1.16, 1.15, and 1.14 patch releases. That means these assertions will likely break because there will be a different number of built-in policies that this expects.

We probably want to remove these `require.Len` assertions and only check whether the anonymous-token-policy was created or not. (I should have written it this way originally 🙃 ):

* If partitionsEnabled:
  * `policies` does not contain `"anonymous-token-policy"` (i.e. `require.NotContains(...)`)
* Else:
  * `policies` does contain `"anonymous-token-policy"`
 
Or we could just use the [Read Policy by Name API](https://developer.hashicorp.com/consul/api-docs/acl/policies#read-a-policy-by-name) to directly check for the anonymous-token-policy.

Reply:
Fixed this